pkg/apis/duck/v1/delivery_types.go

```diff
@@ -155,3 +155,52 @@ type DeliveryStatus struct {
 	// +optional
 	DeadLetterSinkURI *apis.URL `json:"deadLetterSinkUri,omitempty"`
 }
+
+func (d *DeliverySpec) Merge(other *DeliverySpec) *DeliverySpec {
+	if d == nil {
+		return other
+	}
+	if other == nil {
```

Comment:
I think this both checks could be moved outside the function, but if you have to have it as a helper func is ok to leave them here